src/eosjs-webauthn-sig.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @module WebAuthn-Sig
+ */
+// copyright defined in eosjs/LICENSE.txt
+
+import { SignatureProvider, SignatureProviderArgs } from './eosjs-api-interfaces';
+import * as ser from './eosjs-serialize';
+import * as numeric from './eosjs-numeric';
+import { ec } from 'elliptic';
+
+/** Signs transactions using WebAuthn */
+export class WebAuthnSignatureProvider implements SignatureProvider {
+    /** Map public key to credential ID (hex). User must populate this. */
+    public keys = new Map<string, string>();
+
+    /** Public keys that the `SignatureProvider` holds */
+    public async getAvailableKeys() {
+        return Array.from(this.keys.keys());
+    }
+
+    /** Sign a transaction */
+    public async sign(
+        { chainId, requiredKeys, serializedTransaction, serializedContextFreeData }:
+            SignatureProviderArgs,
+    ) {
+        const signBuf = new ser.SerialBuffer();
+        signBuf.pushArray(ser.hexToUint8Array(chainId));
+        signBuf.pushArray(serializedTransaction);
+        if (serializedContextFreeData) {
+            signBuf.pushArray(new Uint8Array(await crypto.subtle.digest('SHA-256', serializedContextFreeData.buffer)));
+        } else {
+            signBuf.pushArray(new Uint8Array(32));
+        }
+        const digest = new Uint8Array(await crypto.subtle.digest('SHA-256', signBuf.asUint8Array().slice().buffer));
+
+        const signatures = [] as string[];
+        for (const key of requiredKeys) {
+            const id = ser.hexToUint8Array(this.keys.get(key));
+            const assertion = await (navigator as any).credentials.get({
```

Comment:
Same problem here with Nodejs, this is browser only.

Reply:
If it doesn't make sense for this SignatureProvider to run in a Node environment, we should update the README with this PR to make it clear this is browser only.